Framework/test/testKafkaTests.cxx

```diff
@@ -0,0 +1,484 @@
+// Copyright 2024 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    testKafkaTests.cxx
+/// \author  Michal Tichak
+/// \brief Tests for kafkaPoller class and StartOfRun and EndOfRun triggers
+///
+/// Detailed description:
+/// We are testing SOR and EOR triggers here so we have all tests that require
+/// Kafka in one place. Tests in this file can be run manually only right now,
+/// as there is no kafka cluster setup as a part of CI/CD for QC.
+/// Change `kafkaCluster` variable that reflects correct kafka cluster that
+/// you want to use for tests. In order to run these tests call from command line:
+/// `o2-qc-test-core [.manual_kafka]`
+///
+/// NOTE: it might be necessary to recreate or purge topic when you are doing
+/// a lot of consecutive tests.
+///
+
+#include <catch_amalgamated.hpp>
+#include <chrono>
+#include <semaphore>
+#include <thread>
+#include "kafka/ProducerRecord.h"
+#include "kafka/Types.h"
+
+#include <QualityControl/KafkaPoller.h>
+#include <QualityControl/Triggers.h>
+#include <kafka/KafkaProducer.h>
+#include <proto/events.pb.h>
+
+// change this url if you are testing on different kafka cluster
+constexpr auto kafkaCluster = "mtichak-flp-1-27.cern.ch:9092";
```

Comment:
perhaps put there some invalid value, you will avoid someone else testing on your instance by accident and making you wonder what happened. Also, it is somewhat more secure to avoid exposing internal endpoints.